light-zk.js/src/utxo.ts

```diff
@@ -216,15 +225,35 @@ export class Utxo {
     this.includeAppData = includeAppData;
     this.transactionVersion = "0";
 
+    if (
+      assets[1].toBase58() === SystemProgram.programId.toBase58() &&
+      !amounts[1].isZero()
+    ) {
+      throw new UtxoError(
+        UtxoErrorCode.POSITIVE_AMOUNT,
+        "constructor",
+        `spl amount cannot be positive, amounts[1] = ${
+          amounts[1] ?? "undefined"
+        }`,
+      );
+    }
     // TODO: make variable length
-    if (assets[1].toBase58() != SystemProgram.programId.toBase58()) {
+    else if (assets[1].toBase58() != SystemProgram.programId.toBase58()) {
       this.assetsCircuit = [
         hashAndTruncateToCircuit(SystemProgram.programId.toBytes()),
         hashAndTruncateToCircuit(this.assets[1].toBytes()),
       ];
-    } else if (this.amounts[0].toString() === "0") {
+    } else if (this.amounts[0].isZero()) {
       this.assetsCircuit = [BN_0, BN_0];
-    } else {
+    }
+    // else if (!this.amounts[0].isZero()) {
+    //   throw new UtxoError(
+    //     UtxoErrorCode.NON_ZERO_AMOUNT,
+    //     "constructor",
+    //     `amount not zero, amounts[0] = ${this.amounts[0] ?? "undefined"}`,
+    //   );
+    // }
```

Comment:
We shouldn't throw an error if the amount is positive since this is normal usage.
Pls remove :)

Reply:
removed ✅